contrib/platform/src/com/sun/jna/platform/WindowUtils.java

```diff
@@ -1269,11 +1270,10 @@ public boolean callback(final HWND hwnd, final Pointer arg1) {
         public String getWindowTitle(final HWND hwnd) {
             final int requiredLength = User32.INSTANCE
                 .GetWindowTextLength(hwnd) + 1;
-            final char[] title = new char[requiredLength];
-            final int length = User32.INSTANCE.GetWindowText(hwnd, title,
-                                                             title.length);
+            NativeWin32String title = new NativeWin32String(requiredLength);
+            User32.INSTANCE.GetWindowText(hwnd, title, title.size());
```

Comment:
So `User32.GetWindowText()` is indeed broken assuming unicode with the `char[]` mapping, but you can't change its API without a version bump.  We need to discuss a workaround.  One Idea I have is explicitly adding the `A` and `W` versions, properly mapped, deprecating the undecorated name mapping, and updating this util to call the appropriate one.  @matthiasblaesing may have a better idea.